repl/src/main/scala/com/cloudera/livy/repl/SparkContextInitializer.scala

```diff
@@ -0,0 +1,123 @@
+      .asInstanceOf[SparkContext]
+    var sqlContext: Object = null
+
+    if (conf.getBoolean("spark.repl.enableHiveContext", false)) {
+      try {
+        val loader = Option(Thread.currentThread().getContextClassLoader)
+          .getOrElse(getClass.getClassLoader)
+        if (loader.getResource("hive-site.xml") == null) {
+          warn("livy.repl.enableHiveContext is true but no hive-site.xml found on classpath.")
+        }
+
+        sqlContext = Class.forName("org.apache.spark.sql.hive.HiveContext")
+          .getConstructor(classOf[SparkContext]).newInstance(sparkContext).asInstanceOf[Object]
+        info("Created sql context (with Hive support).")
+      } catch {
+        case _: NoClassDefFoundError =>
+          sqlContext = Class.forName("org.apache.spark.sql.SQLContext")
+            .getConstructor(classOf[SparkContext]).newInstance(sparkContext).asInstanceOf[Object]
+          info("Created sql context.")
+      }
+    } else {
+      sqlContext = Class.forName("org.apache.spark.sql.SQLContext")
+        .getConstructor(classOf[SparkContext]).newInstance(sparkContext).asInstanceOf[Object]
+      info("Created sql context.")
+    }
+
+    bind("sc", "org.apache.spark.SparkContext", sparkContext, List("""@transient"""))
+    bind("sqlContext", sqlContext.getClass.getCanonicalName, sqlContext, List("""@transient"""))
+
+    execute("import org.apache.spark.SparkContext._")
+    execute("import sqlContext.implicits._")
+    execute("import sqlContext.sql")
+    execute("import org.apache.spark.sql.functions._")
+  }
+
+  private def spark2CreateContext(conf: SparkConf): Unit = {
+    val sparkClz = Class.forName("org.apache.spark.sql.SparkSession$")
+    val sparkObj = sparkClz.getField("MODULE$").get(null)
+
+    val builderMethod = sparkClz.getMethod("builder")
+    val builder = builderMethod.invoke(sparkObj)
+    builder.getClass.getMethod("config", classOf[SparkConf]).invoke(builder, conf)
+
+    var spark: Object = null
+    if (conf.get("spark.sql.catalogImplementation", "hive").toLowerCase == "hive") {
+      if (sparkClz.getMethod("hiveClassesArePresent").invoke(sparkObj).asInstanceOf[Boolean]) {
+        builder.getClass.getMethod("enableHiveSupport").invoke(builder)
+        spark = builder.getClass.getMethod("getOrCreate").invoke(builder)
+        info("Created Spark session with Hive support")
+      } else {
+        builder.getClass.getMethod("config", classOf[String], classOf[String])
+          .invoke(builder, "spark.sql.catalogImplementation", "in-memory")
+        spark = builder.getClass.getMethod("getOrCreate").invoke(builder)
+        info("Created Spark session")
+      }
+    } else {
+      spark = builder.getClass.getMethod("getOrCreate").invoke(builder)
+      info("Created Spark session")
+    }
+
+    sparkContext = spark.getClass.getMethod("sparkContext").invoke(spark)
+      .asInstanceOf[SparkContext]
+
+    bind("spark", spark.getClass.getCanonicalName, spark, List("""@transient"""))
```

Comment:
Do we really need such hints? Since spark-shell is facing to the end-user, while interpreter in Livy isn't.